src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -183,6 +185,29 @@ const getObjectsToFetch = (queue: BatchQueueEntry[]): ObjectTypeAndId[] => {
 export class SavedObjectsClient {
   private http: HttpSetup;
   private batchQueue: BatchQueueEntry[];
+  /**
+   * The currentWorkspaceId may be undefiend when workspace plugin is not enabled.
+   */
+  private currentWorkspaceId: string | undefined;
+
+  /**
+   * Check if workspaces field present in given options, if so, overwrite the current workspace id.
+   * @param options
+   * @returns
+   */
+  private formatWorkspacesParams(options: {
+    workspaces?: SavedObjectsCreateOptions['workspaces'];
+  }) {
+    const currentWorkspaceId = this.currentWorkspaceId;
+    let finalWorkspaces;
+    if (options.hasOwnProperty('workspaces')) {
+      finalWorkspaces = options.workspaces;
+    } else if (typeof currentWorkspaceId === 'string') {
+      finalWorkspaces = [currentWorkspaceId];
+    }
+
+    return finalWorkspaces;
+  }
```

Comment:
Nit,  not a big deal, but would this look a bit cleaner?
```suggestion
  private formatWorkspacesParams(options: {
    workspaces?: SavedObjectsCreateOptions['workspaces'];
  }): {workspaces: string[]} | {} {
    const currentWorkspaceId = this.currentWorkspaceId;
    let finalWorkspaces;
    if (options.hasOwnProperty('workspaces')) {
      finalWorkspaces = options.workspaces;
    } else if (typeof currentWorkspaceId === 'string') {
      finalWorkspaces = [currentWorkspaceId];
    }
    
    if (finalWorkspaces) {
      return {workspaces: finalWorkspaces}
    }

    return {};
  }
  
  // then you could do
  const workspaceParams = this.formatWorkspacesParams(options);
```

Reply:
Yes, and reduce more duplicate code. Done for that.